spec/Section 3 -- Type System.md

```diff
@@ -1448,6 +1480,28 @@ define arguments or contain references to interfaces and unions, neither of
 which is appropriate for use as an input argument. For this reason, input
 objects have a separate type in the system.
 
+**Oneof Input Objects**
+
+Oneof Input Objects are a special variant of Input Objects where the type
+system asserts that exactly one of the fields must be set and non-null, all
+others being omitted. This is useful for representing situations where an input
+may be one of many different options.
+
+When using the type system definition language, the `@oneOf` directive is used
+to indicate that an Input Object is a Oneof Input Object (and thus requires
+exactly one of its field be provided):
+
+```graphql
+input UserUniqueCondition @oneOf {
+  id: ID
+  username: String
+  organizationAndEmail: OrganizationAndEmailInput
+}
+```
+
+In schema introspection, the `__Type.oneField` field will return {true} for
+Oneof Input Objects, and {false} for all other Input Objects.
```

Comment:
```suggestion
Oneof Input Objects, and {false} for all other types.
```
?

Also what abour returning `null` ? `oneField` is `Boolean` compared to `oneArgument` that is `Boolean!`.

Reply:
`__Type.oneField` follows the pattern of all other `__Type` fields: it's nullable. The concept of `oneField` doesn't make sense for a union for example, so having `oneField: false` doesn't make sense for a union; hence nullable.

`__Field.oneArgument`, on the other hand, makes sense for all fields, so it's perfectly fine to be required.

Reply:
OK